tests/test_device_properties.py

```diff
@@ -0,0 +1,56 @@
+"""Tests for validating device properties for different basic data types."""
+from nidaqmx.constants import BusType, TriggerUsage
+
+
+class TestDevicePropertyDataTypes(object):
+    """Contains a collection of pytest tests.
+
+    This validates the property getter,setter and deleter methods for different
+    data types of device properties.
+    """
+
+    def test_boolean_property(self, bridge_device):
+        """Test for validating the device property of boolean type."""
+        # The PXIe-4331 supports analog triggering
+        assert bridge_device.anlg_trig_supported
+
+    def test_float_property(self, bridge_device):
+        """Test for validating the device property of float type."""
+        # The ai max single channel rate of the NI PXIe-4331 device is 102400.0
+        assert bridge_device.ai_max_single_chan_rate == 102400.0
+
+    def test_uint_property(self, bridge_device):
+        """Test for validating the device property of uint type."""
+        assert bridge_device.product_num == 1957283012
+
+    def test_string_property(self, bridge_device):
+        """Test for validating a device property of string type."""
+        assert bridge_device.product_type == "PXIe-4331"
```

Comment:
The `bridge_device` fixture matches against the first device that supports `UsageTypeAI.BRIDGE`: https://github.com/ni/nidaqmx-python/blob/master/tests/conftest.py#L77

If you have another bridge device in your system, these tests may use the wrong device and fail.

Some ideas:
- Write a local fixture that returns `nidaqmx.system.Device("bridgeTester")`
- Skip if the device's name isn't "bridgeTester"

Reply:
I have included a common fixture `test_device`, which could get the device based on name.
This fixture is now used for all device tests to get the bridgeTester device based on name.
I added a common fixture, as I feel this might be required by other tests that needs to get a device based on the name.

Reply:
Good, the new fixture will provide helpful errors when the specified device is missing.